### PR TITLE
zigbee: Fixes Zigbee weather station application and optimize calling ZBOSS API in NCS [KRKNWK-14018]

### PR DESCRIPTION
**Pull request by sebastiandraus**
- Fixes Zigbee weather station application by removing incorrect buffer freeing
- Optimize calling ZBOSS API in NCS platform by skipping unnecessary processing by workqueue

### CODE REVIEW

**Review by adsz-nordic — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -146,6 +146,7 @@ Thingy:53 Zigbee weather station
 --------------------------------
 
 * Added new :ref:`zigbee_weather_station_app` application.
+* Fixed an issue where buffer is being freed incorrectly.
```
Comment (kittydepa) [inline doc/nrf/releases/release-notes-changelog.rst:149]:
```suggestion
* Fixed an issue where the buffer was being freed incorrectly.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -694,6 +695,8 @@ Libraries for Zigbee
 
       * Crypto library used for performing software AES ecryption.
         Now, the :ref:`nrfxlib:nrf_oberon_readme` is used instead of the Tinycrypt library.
+      * Optimize calling ZBOSS API in NCS platform.
+        Now, if ZBOSS API is called from ZBOSS thread context, processing by workqueue is skipped.
```
Comment (kittydepa) [inline doc/nrf/releases/release-notes-changelog.rst:699]:
```suggestion
       If the ZBOSS API is called in the ZBOSS thread context, processing by workqueue is now skipped.
```
Please make sure that this means the same thing after my suggestion.

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -694,6 +695,8 @@ Libraries for Zigbee
 
       * Crypto library used for performing software AES ecryption.
         Now, the :ref:`nrfxlib:nrf_oberon_readme` is used instead of the Tinycrypt library.
+      * Optimize calling ZBOSS API in NCS platform.
+        If the ZBOSS API is called in the ZBOSS thread context, processing by workqueue is now skipped.
```
Comment (kittydepa) [inline doc/nrf/releases/release-notes-changelog.rst:699]: Sorry, I missed this the first time! One tiny edit:
```suggestion
        If the ZBOSS API is called in the ZBOSS thread context, processing by the workqueue is now skipped.
```

Comment (sebastiandraus) [reply doc/nrf/releases/release-notes-changelog.rst]: Done!

**Review by kittydepa — APPROVED** [review]

**Comment by sebastiandraus** [comment]
Rebased at newest main branch because of Jeknkins failing for some reason

**Pull request merged**